src/coreclr/jit/lower.cpp

```diff
@@ -7903,10 +7903,16 @@ void Lowering::LowerBlockStoreAsHelperCall(GenTreeBlk* blkNode)
         else
         {
             assert(data->OperIs(GT_LCL_VAR, GT_LCL_FLD));
+            const bool isLclVar = data->OperIs(GT_LCL_VAR);
 
             // Convert local to LCL_ADDR
             data->ChangeOper(GT_LCL_ADDR);
             data->ChangeType(TYP_I_IMPL);
+            if (isLclVar)
+            {
+                // Clear offset if it wasn't a LCL_FLD
+                data->AsLclFld()->SetLclOffs(0);
+            }
```

Comment:
Might as well unconditionally propagate the offset then, which would be a bit simpler code.

Reply:
Yeah, that is simpler, changed 🙂